libcfnet/stat_cache.c

```diff
@@ -165,7 +165,11 @@ int cf_remote_stat(AgentConnection *conn, bool encrypt, const char *file,
 
         tosend = cipherlen + CF_PROTO_OFFSET;
 
-        if(tosend > sizeof(sendbuffer))
+        if (tosend < 0)
+        {
+            ProgrammingError("cf_remote_stat: tosend (%d) < 0", tosend);
+        }
+        else if((unsigned int) tosend > sizeof(sendbuffer))
```

Comment:
You might as well cast it to an `unsigned long`, as the compiler will have to do this anyways.

> Otherwise, if both operands have signed integer types or both have unsigned integer types, the operand with the type of lesser integer conversion rank is converted to the type of the operand with greater rank.
 - C99 section 6.3.1.8

Reply:
Oh, this is me? Correction: "I might as well ... "